providers/exoscale/exoscaleProvider.go

```diff
@@ -0,0 +1,203 @@
+package exoscale
+
+import (
+	"encoding/json"
+	"strings"
+
+	"github.com/exoscale/egoscale"
+	"github.com/pkg/errors"
+
+	"github.com/StackExchange/dnscontrol/models"
+	"github.com/StackExchange/dnscontrol/providers"
+	"github.com/StackExchange/dnscontrol/providers/diff"
+)
+
+type exoscaleProvider struct {
+	client *egoscale.Client
+}
+
+var defaultNameServerNames = []string{
+	"ns1.exoscale.net",
+	"ns1.exoscale.ch",
+	"ns1.exoscale.com",
+	"ns1.exoscale.io",
+}
+
+// NewExoscale creates a new Exoscale DNS provider.
+func NewExoscale(m map[string]string, metadata json.RawMessage) (providers.DNSServiceProvider, error) {
+	endpoint, apiKey, secretKey := m["dns-endpoint"], m["api-key"], m["secret-key"]
+
+	return &exoscaleProvider{client: egoscale.NewClient(endpoint, apiKey, secretKey)}, nil
+}
+
+var features = providers.DocumentationNotes{
+	providers.CanUseAlias:            providers.Can(),
+	providers.CanUseCAA:              providers.Can(),
+	providers.CanUsePTR:              providers.Can(),
+	providers.CanUseSRV:              providers.Can(),
+	providers.CanUseTLSA:             providers.Cannot(),
+	providers.DocCreateDomains:       providers.Cannot(),
+	providers.DocDualHost:            providers.Cannot(),
+	providers.DocOfficiallySupported: providers.Cannot(),
+}
+
+func init() {
+	providers.RegisterDomainServiceProviderType("EXOSCALE", NewExoscale, features)
+}
+
+// EnsureDomainExists returns an error if domain doesn't exist.
+func (c *exoscaleProvider) EnsureDomainExists(domain string) error {
+	_, err := c.client.GetDomain(domain)
+	if err != nil {
+		_, err := c.client.CreateDomain(domain)
+		if err != nil {
+			return err
+		}
+	}
+	return err
+}
+
+// GetNameservers returns the nameservers for domain.
+func (c *exoscaleProvider) GetNameservers(domain string) ([]*models.Nameserver, error) {
+	return models.StringsToNameservers(defaultNameServerNames), nil
+}
+
+// GetDomainCorrections returns a list of corretions for the  domain.
+func (c *exoscaleProvider) GetDomainCorrections(dc *models.DomainConfig) ([]*models.Correction, error) {
+	dc.Punycode()
+
+	records, err := c.client.GetRecords(dc.Name)
+	if err != nil {
+		return nil, err
+	}
+
+	existingRecords := make([]*models.RecordConfig, len(records))
```

Comment:
This creates an array of length "len(records)". If you append to it, you add more records. Since you skip some records, just don't specify initial length.

Reply:
thx